File: docs/APIs-and-SDKs/SDK-Documentation/complete-example/php/completeExample.php
<?php

use \ABSmartly\SDK\SDK;
use \ABSmartly\SDK\Context\ContextConfig;

// Initialize the SDK
$sdk = SDK::createWithDefaults(
    endpoint: 'https://your-company.absmartly.io/v1',
    apiKey: 'YOUR-API-KEY',
    environment: 'production',
    application: 'website',
);

// Create a context for this user
$contextConfig = new ContextConfig();
$contextConfig->setUnit('user_id', 'user-12345');
$context = $sdk->createContext($contextConfig);

// Check which variant the user is in
$treatment = $context->getTreatment('homepage_banner_experiment');

if ($treatment === 0) {
    // Variant A (control): show the existing banner
    showBanner('Welcome back!');
} else {
    // Variant B: show a personalized banner
    showBanner('Welcome back, we have new deals for you!');
}

// Use a variable set in the Web Console for more flexibility
$buttonColor = $context->getVariableValue('button.color', 'blue');
setButtonColor($buttonColor);

// Track when the user completes a key action
$context->track('cta_click', (object) ['page' => 'homepage']);

// Track a purchase with revenue data
$context->track('purchase', (object) [
    'revenue' => 49.99,
    'item_count' => 3,
]);

// Close the context to flush remaining events
$context->close();

File: docs/APIs-and-SDKs/SDK-Documentation/complete-example/swift/completeExample.swift
import ABSmartly

// Initialize the SDK
let clientConfig = ClientConfig(
    apiKey: "YOUR-API-KEY",
    application: "website",
    endpoint: "https://your-company.absmartly.io/v1",
    environment: "production"
)
let client = try DefaultClient(config: clientConfig)
let sdkConfig = ABSmartlyConfig.create()
sdkConfig.client = client
let sdk = try ABSmartlySDK(config: sdkConfig)

// Create a context for this user
let contextConfig = ContextConfig()
contextConfig.setUnit(unitType: "user_id", uid: "user-12345")
let context = sdk.createContext(config: contextConfig)

context.waitUntilReady().done { context in
    // Check which variant the user is in
    let treatment = context.getTreatment("homepage_banner_experiment")

    if treatment == 1 {
        // Variant B: show a personalized banner
        showBanner("Welcome back, we have new deals for you!")
    } else {
        // Variant A (control): show the existing banner
        showBanner("Welcome back!")
    }

    // Use a variable set in the Web Console for more flexibility
    let buttonColor = context.getVariableValue("button.color", defaultValue: "blue")
    setButtonColor(buttonColor)

    // Track when the user completes a key action
    context.track("cta_click", properties: ["page": "homepage"])

    // Track a purchase with revenue data
    context.track("purchase", properties: [
        "revenue": 49.99,
        "item_count": 3
    ])

    // Close the context when done to flush remaining events
    context.close()
}
